src/app/components/files/viewer/component.ts

```diff
@@ -24,20 +24,28 @@ export class FileViewerComponent implements OnInit {
     this.updateImages();
   }
 
+  private makeImageComponentStruct(image) {
+    return {
+      id: image.id,
+      src: this.assets.get(image.id)
+    }
+  }
+
   updateImages() {
     const url = 'files?filter[uploaded_by][_eq]=$CURRENT_USER'
     this.api.get(url).subscribe(
       result => {
-        this.images = result.data.map(image => {
-          return {
-            id: image.id,
-            src: this.assets.get(image.id),
-          }
-        }) 
+        this.images = result.data.map(image => this.makeImageComponentStruct(image)) 
       }
     );
   }
 
+  addImage(image) {
+    if (image) {
+      this.images.unshift(this.makeImageComponentStruct(image))
+    }
+  }
+
```

Comment:
I think this is very clever, but rather than directly append to this list, I would rather re-update the entire list from the database

I'm not worried about database load, so updating the entire list will allow the parent `uploader` component to remain agnostic to the image data managed by the `viewer` component.